src/apps/core/submodules/alerts/alerts.js

```diff
@@ -16,10 +16,48 @@ define(function(require) {
 		 * Trigger the alerts pulling process from API.
 		 */
 		alertsRender: function() {
-			var self = this;
+			var self = this,
+				initTemplate = function initTemplate(alerts) {
+					var alertGroups = self.alertsFormatData({ data: alerts }),
+						alertCount = _.reduce(alertGroups, function(count, alertGroup) {
+							return count + alertGroup.length;
+						}, 0),
+						dataTemplate = {
+							showAlertCount: alertCount > 0,
+							alertCount: alertCount > 9 ? '9+' : alertCount.toString()
+						},
+						$template = $(self.getTemplate({
+							name: 'nav',
+							data: dataTemplate,
+							submodule: 'alerts'
+						}));
+
+					monster.ui.tooltips($template);
+
+					self.alertsBindEvents({ template: $template });
+
+					return $template;
+				},
+				renderTemplate = function renderTemplate(alerts) {
+					var $navLinks = $('#main_topbar_nav'),
+						$topbarAlert = $navLinks.find('#main_topbar_alert'),
+						$template = initTemplate(alerts);
+
+					if ($topbarAlert.length === 0) {
+						$template.insertBefore($navLinks.find('#main_topbar_signout'));
+					} else {
+						$topbarAlert.replaceWith($template);
+					}
+				};
 
 			monster.waterfall([
```

Comment:
`monster.ui@waterfall` is not needed here since your functions do not require results from the previous ones.
```suggestion
			monster.series([
```